test/OpenTelemetry.Instrumentation.AspNet.Tests/HttpInListenerTests.cs

```diff
@@ -46,21 +46,23 @@ public void Dispose()
         }
 
         [Theory]
-        [InlineData("http://localhost/", 0, null, "TraceContext")]
-        [InlineData("http://localhost/", 0, null, "TraceContext", true)]
-        [InlineData("https://localhost/", 0, null, "TraceContext")]
-        [InlineData("http://localhost:443/", 0, null, "TraceContext")] // Test http over 443
-        [InlineData("https://localhost:80/", 0, null, "TraceContext")] // Test https over 80
-        [InlineData("http://localhost:80/Index", 1, "{controller}/{action}/{id}", "TraceContext")]
-        [InlineData("https://localhost:443/about_attr_route/10", 2, "about_attr_route/{customerId}", "TraceContext")]
-        [InlineData("http://localhost:1880/api/weatherforecast", 3, "api/{controller}/{id}", "TraceContext")]
-        [InlineData("https://localhost:1843/subroute/10", 4, "subroute/{customerId}", "TraceContext")]
-        [InlineData("http://localhost/api/value", 0, null, "TraceContext", false, "/api/value")] // Request will be filtered
-        [InlineData("http://localhost/api/value", 0, null, "TraceContext", false, "{ThrowException}")] // Filter user code will throw an exception
-        [InlineData("http://localhost/api/value/2", 0, null, "CustomContextMatchParent")]
-        [InlineData("http://localhost/api/value/2", 0, null, "CustomContextNonmatchParent")]
-        [InlineData("http://localhost/api/value/2", 0, null, "CustomContextNonmatchParent", false, null, true)]
+        [InlineData("http://localhost/", "http://localhost/", 0, null, "TraceContext")]
+        [InlineData("http://localhost/", "http://localhost/", 0, null, "TraceContext", true)]
+        [InlineData("https://localhost/", "https://localhost/", 0, null, "TraceContext")]
+        [InlineData("https://localhost/", "https://user:pass@localhost/", 0, null, "TraceContext")] // Test URL sanitization
```

Comment:
Could we add test cases to cover all the parts that would make up the new Uri such as scheme, authority, path and query, fragment etc.?

How about adding two unit tests like these which would ensure that `Uri.PathAndQuery` and `Uri.Fragment` are concatenated?

**Url:** `https://user:password@localhost:443/Home/Index.html#search`
**ExpectedUrl:** `https://localhost:443/Home/Index.html#search`

And one for http
**Url:** `http://user:password@localhost:443/Home/Index.html#search`
**ExpectedUrl:** `http://localhost:443/Home/Index.html#search`

Reply:
I did my best to address it here: https://github.com/open-telemetry/opentelemetry-dotnet/pull/1961/commits/5f517db45302cc7be9bd293eec01bb9655602abe and here: https://github.com/open-telemetry/opentelemetry-dotnet/pull/1961/commits/275d431b234ccf42d21677f56be5e79cb521772e

Reply:
Added a [comment](https://github.com/open-telemetry/opentelemetry-dotnet/pull/1961/files#r609863384).